connectivity/check/action.go

```diff
@@ -125,8 +125,40 @@ func (a *Action) Destination() TestPeer {
 // This method is to be called from a Scenario implementation.
 func (a *Action) Run(f func(*Action)) {
 	a.Logf("[.] Action [%s]", a)
+
+	// Emit unbuffered progress indicator.
 	a.test.progress()
 
+	// Only perform flow validation if a Hubble Relay connection is available.
+	if a.test.ctx.params.Hubble {
+		// Channel for the flow listener to notify us when ready.
+		ready := make(chan bool, 1)
+
+		// TODO(timo): Use an actual context that can be cancelled by the user.
+		// `Run()` should take context.
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// Start flow listener in the background.
+		go func() {
+			if err := a.followFlows(ctx, ready); err != nil {
```

Comment:
> If we also had a channel for the flows from the follow-listener to the main goroutine we could drain that channel in the test loop and try validation after each such drain loop instead retrying on a ticker?

that's the plan i think. it will change the overall structure of the flow validation, so we wanted to do that in a separate followup PR to limit the scope of this PR.

Reply:
Yes, the changes here are just the first step of this work.